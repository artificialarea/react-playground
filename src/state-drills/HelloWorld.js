import React from 'react';

// src: https://courses.thinkful.com/react-v1/checkpoint/9#assignment
// alt solution: https://github.com/tomatau/state-drills/blob/master/src/state-drills/HelloWorld.js

class HelloWorld extends React.Component {
  state = {
    who: 'who?',
  }

  handleButtonClick = (event) => {
    this.setState({who: event.target.value})
  }

  render() {
    return (
      <div className="greetings">

        <p>Hello, {this.state.who}</p>

        <button value="world!"
          onClick={this.handleButtonClick}
        >
          World
        </button>
        <button value="friend!"
          onClick={this.handleButtonClick}
        >
          Friend
        </button>
        <button value="React!"
          onClick={this.handleButtonClick}
        >
          React
        </button>
      </div>
    )
  }
}

export default HelloWorld;