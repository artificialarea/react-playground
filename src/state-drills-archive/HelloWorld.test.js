import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import HelloWorld from './HelloWorld';

describe('<HelloWorld />', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<HelloWorld />, div)
    ReactDOM.unmountComponentAtNode(div)  
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(<HelloWorld />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});