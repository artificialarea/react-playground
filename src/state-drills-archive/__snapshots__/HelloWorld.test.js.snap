// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelloWorld /> renders UI as expected 1`] = `
<div
  className="greetings"
>
  <p>
    Hello, 
    who?
  </p>
  <button
    onClick={[Function]}
    value="world!"
  >
    World
  </button>
  <button
    onClick={[Function]}
    value="friend!"
  >
    Friend
  </button>
  <button
    onClick={[Function]}
    value="React!"
  >
    React
  </button>
</div>
`;
